onnxruntime/core/optimizer/initializer.h

```diff
@@ -589,12 +595,15 @@ class Initializer final {
   }
 
  private:
+  static Status ReadExternalRawData(
+      const ONNX_NAMESPACE::TensorProto& tensor_proto, const Path& model_path, std::vector<char>& raw_data);
+
```

Comment:
This might not be required yet, but I would have expected if something was stored using external data, the replacement or derivative data created by the optimizer would also likely need to be stored using external data. For that to happen this class would need to store info about where the data was loaded from. Can be separate PR to support that when required. 

Side note: May be best for the behavior around whether to use external data or not for the updated/new initializers to be controlled by the user, and we need a good way for such low-level configuration options to be defined. Adding items to SessionOptions for such things would quickly get out of control (e.g. I have two settings specific to BFCArena I want to add).

Reply:
Ok, let's do that in a separate PR when needed.